editoast/src/generated_data/mod.rs

```diff
@@ -67,22 +77,29 @@ pub trait GeneratedData {
 
 /// Refresh all the generated data of a given infra
 pub async fn refresh_all(
-    conn: &mut PgConnection,
+    db_pool: crate::Data<crate::DbPool>,
     infra: i64,
     infra_cache: &InfraCache,
 ) -> Result<()> {
-    TrackSectionLayer::refresh(conn, infra, infra_cache).await?;
-    SpeedSectionLayer::refresh(conn, infra, infra_cache).await?;
-    SignalLayer::refresh(conn, infra, infra_cache).await?;
-    SwitchLayer::refresh(conn, infra, infra_cache).await?;
-    BufferStopLayer::refresh(conn, infra, infra_cache).await?;
-    CatenaryLayer::refresh(conn, infra, infra_cache).await?;
-    DetectorLayer::refresh(conn, infra, infra_cache).await?;
-    OperationalPointLayer::refresh(conn, infra, infra_cache).await?;
-    TrackSectionLinkLayer::refresh(conn, infra, infra_cache).await?;
-    LPVPanelLayer::refresh(conn, infra, infra_cache).await?;
-    ErrorLayer::refresh(conn, infra, infra_cache).await?;
-    NeutralSectionLayer::refresh(conn, infra, infra_cache).await?;
+    // The other layers depend on track section layer.
+    // We must wait until its completion before running the other requests in parallel
+    TrackSectionLayer::refresh_pool(db_pool.clone(), infra, infra_cache).await?;
+    log::debug!("⚙️ Infra {infra}: track section layer is generated");
+    let mut conn = db_pool.get().await?;
+    sql_query("analyze").execute(&mut conn).await?;
```

Comment:
Should we add a comment to explain why an analyse is required here?

Reply:
I added a comment explaining the situation